core/src/main/java/org/elasticsearch/search/suggest/phrase/PhraseSuggestionBuilder.java

```diff
@@ -417,7 +418,7 @@ public Script collateQuery() {
      */
     public PhraseSuggestionBuilder collateParams(Map<String, Object> collateParams) {
         Objects.requireNonNull(collateParams, "collate parameters cannot be null.");
-        this.collateParams = new HashMap<>(collateParams);
+        this.collateParams = new LinkedHashMap<>(collateParams);
```

Comment:
I think if we need to be careful about ordering the parameters for serialization we should do it during serialization. In this case we'll just get them in whatever order the `collateParams` iterates which I imagine is fairly inconsistent. My gut says `HashMaps` are going to be the most common parameter here and they will iterate in a vaguely consistent order. Like, the same map contents won't always have a consistent order because they could have a different underlying hash table size.

So I think the sorting you do is during writing is better. No matter what how crazy we are with the map we always will get in order writes.